Problem with bakta annotations (gene being longer than the contig it is on)

Hi there!

I was trying to run ppanggolin (v2.2.4) on a collection of P. aeruginosa genomes, using bakta's gff3 annotations, and got the following error (please see below). Please bear in mind that I renamed the protein IDs, but that is not the problem, since I ran ppannggolin against the "original" bakta annotations and the error continues. Panaroo has an option to remove invalid genes, but I guess there's no similar option in ppanggolin?

I noticed there are a couple of issues referring to a similar topic here, and from what I could understand, the problem was already fixed. So not clear for me why I'm still seeing this.

Thanks a lot!

```
3%|▎         | 302/11655 [00:15<09:29, 19.94file/s]
concurrent.futures.process._RemoteTraceback:
"""
Traceback (most recent call last):
  File "/gxfs_work/cau/sunzm592/.conda/envs/ppanggolinv2.2.4/lib/python3.12/site-packages/ppanggolin/annotate/annotate.py", line 1427, in read_anno_file
    org, has_fasta = read_org_gff(
                     ^^^^^^^^^^^^^
  File "/gxfs_work/cau/sunzm592/.conda/envs/ppanggolinv2.2.4/lib/python3.12/site-packages/ppanggolin/annotate/annotate.py", line 1223, in read_org_gff
    rna.add_sequence(get_dna_sequence(contig_sequences[[contig.name](http://contig.name/)], rna))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gxfs_work/cau/sunzm592/.conda/envs/ppanggolinv2.2.4/lib/python3.12/site-packages/ppanggolin/annotate/synta.py", line 488, in get_dna_sequence
    assert highest_position <= len(
           ^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Coordinates of gene GCF_000790035.1_tmRNA_0070 exceed length of the contig. Gene coordinates [[(451497, 451846](callto:(451497, 451846))] vs contig length 451844

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/gxfs_work/cau/sunzm592/.conda/envs/ppanggolinv2.2.4/lib/python3.12/concurrent/futures/process.py", line 264, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gxfs_work/cau/sunzm592/.conda/envs/ppanggolinv2.2.4/lib/python3.12/site-packages/ppanggolin/annotate/annotate.py", line 1431, in read_anno_file
    raise Exception(
Exception: Reading the gff3 file 'Pa_only_genomes/GCF_000790035.1_renamed/GCF_000790035.1_renamed.gff3' raised an error. Coordinates of gene GCF_000790035.1_tmRNA_0070 exceed length of the contig. Gene coordinates [[(451497, 451846](callto:(451497, 451846))] vs contig length 451844
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gxfs_work/cau/sunzm592/.conda/envs/ppanggolinv2.2.4/bin/ppanggolin", line 10, in <module>
    sys.exit(main())
             ^^^^^^
  File "/gxfs_work/cau/sunzm592/.conda/envs/ppanggolinv2.2.4/lib/python3.12/site-packages/ppanggolin/main.py", line 283, in main
    ppanggolin.workflow.all.launch(args)
  File "/gxfs_work/cau/sunzm592/.conda/envs/ppanggolinv2.2.4/lib/python3.12/site-packages/ppanggolin/workflow/all.py", line 464, in launch
    launch_workflow(args, panrgp=True, panmodule=True)
  File "/gxfs_work/cau/sunzm592/.conda/envs/ppanggolinv2.2.4/lib/python3.12/site-packages/ppanggolin/workflow/all.py", line 71, in launch_workflow
    read_annotations(
  File "/gxfs_work/cau/sunzm592/.conda/envs/ppanggolinv2.2.4/lib/python3.12/site-packages/ppanggolin/annotate/annotate.py", line 1563, in read_annotations
    org, has_dna_sequence = future.result()
                            ^^^^^^^^^^^^^^^
  File "/gxfs_work/cau/sunzm592/.conda/envs/ppanggolinv2.2.4/lib/python3.12/concurrent/futures/_base.py", line 456, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/gxfs_work/cau/sunzm592/.conda/envs/ppanggolinv2.2.4/lib/python3.12/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
Exception: Reading the gff3 file 'Pa_only_genomes/GCF_000790035.1_renamed/GCF_000790035.1_renamed.gff3' raised an error. Coordinates of gene GCF_000790035.1_tmRNA_0070 exceed length of the contig. Gene coordinates [[(451497, 451846](callto:(451497, 451846))] vs contig length 451844
```

--- a/ppanggolin/annotate/annotate.py
+++ b/ppanggolin/annotate/annotate.py
@@ -4,6 +4,7 @@
 import argparse
 import logging
 from concurrent.futures import ProcessPoolExecutor
+from itertools import chain
 
 from multiprocessing import get_context
 import os
@@ -37,7 +38,6 @@ from ppanggolin.utils import (
     check_input_files,
     has_non_ascii,
     replace_non_ascii,
-    check_tools_availability,
 )
 from ppanggolin.formats import write_pangenome
 from ppanggolin.metadata import Metadata
@@ -1359,7 +1359,8 @@ def correct_putative_overlaps(contigs: Iterable[Contig]):
     """
 
     for contig in contigs:
-        for gene in contig.genes:
+
+        for gene in chain(contig.genes, contig.RNAs):
             if gene.stop > len(contig):
                 # Adjust gene coordinates to handle circular contig
                 gene.start = 1  # Start gene at the beginning of the contig

